File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<char> c;
        int n = s.size(), ans=0,i=0,j=0;
        while(i<n && j<n){
            if(c.find(s[j])==c.end()){
                c.insert(s[j++]);
                ans = max(ans,j-i);
            }else{
                c.erase(s[i++]);
            }
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s)==0:
            return 0
        mx=0
        s=list(s)
        first=0
        second=0
        count=1
        for i in range(1,len(s)):
            if s[i] not in s[first:i]:
                count+=1
            else:
                mx=max(mx,count)
                count=i-s[first:i].index(s[i])-len(s[:i])+len(s[first:i])
                first=s[first:i].index(s[i])+1+len(s[:first])
        return max(mx,count)